crates/solidity/outputs/npm/build/src/main.rs

```diff
@@ -1,21 +1,18 @@
 use anyhow::Result;
-use codegen_schema::types::LanguageDefinition;
-use codegen_syntax::SyntaxGeneratorExtensions;
+use codegen_grammar::Grammar;
+use codegen_parser_generator::code_generator::CodeGenerator;
 use infra_utils::cargo::CargoWorkspace;
-use solidity_language::SolidityLanguageExtensions;
+use solidity_language::GrammarConstructor;
 
 // Instead of the soure crate calling codegen APIs directly, it invokes this binary, which in turn calls the codegen APIs.
 // This indirection is needed because:
 //
 // 1) We want to run codegen, even if there are git conflicts in output files (Cargo fails to build in that case).
 // 2) In order to avoid build graph deadlocks, this binary uses its own local `target` folder here.
+// 3) We want to avoid having dependencies from the source crate to codegen crates.
 //
 fn main() -> Result<()> {
-    // Rerun whenever the crate changes:
-    use solidity_npm_build as _;
-
-    let language = LanguageDefinition::load_solidity()?;
-    let crate_dir = CargoWorkspace::locate_source_crate("solidity_npm_crate")?;
-
-    return language.generate_legacy_napi_lib_sources(&crate_dir);
+    let crate_dir = CargoWorkspace::locate_source_crate("slang_solidity")?;
```

Comment:
looks like the npm build crate is generating sources into the rust crate (not the npm one). In that case, what would be responsible for `crates/solidity/outputs/npm/crate/src/generated`?

Reply:
doh! Lost as I was trying to fix the multiple-generate-the-delete problem